gpytorch/variational/variational_strategy.py

```diff
@@ -93,6 +93,11 @@ def forward(self, x, inducing_points, inducing_values, variational_inducing_cova
         induc_data_covar = full_covar[..., :num_induc, num_induc:].evaluate()
         data_data_covar = full_covar[..., num_induc:, num_induc:]
 
+        # If inducing points are not batched, no need to work with batched covariance
+        # Not doing so can cause nasty shape incompatibilies when evaluating with different batch shapes
+        if self.inducing_points.ndim <= 2 and len(induc_induc_covar.size()) > 2:
+            induc_induc_covar = induc_induc_covar[0]
```

Comment:
Better might be something like:
```suggestion
            batch_ind = [0] * (len(induc_induc_covar.size() - 2)
            induc_induc_covar = induc_induc_covar[*batch_ind]
```
for compatability with multibatch

Reply:
Ok this is more painful than that, especially when having to deal with the issue that the inducing point covar is cashed for some batch size, but then is evaluated on another batch size. I went with the pythonic route to catch the RuntimeError and clearing the cache. It's not great, but it works and is much more transparent than any index juggling could be.

Reply:
argh, actually this is also not the solution - say you batch evaluate sth and then you evaluate it in non-batch mode with some other test data, that output will now have batch dimensions since the batch dimensions of the cached cholesky factor broadcast :(

Seems like it is necessary to explicitly check whether the batch shape changes.